recipe/meta.yaml

```
@@ -1,13 +1,13 @@
 {% set python_min = "3.10" %}
```

```suggestion
{% set python_min = "3.11" %}
```

Bumping to match [requirements](https://github.com/pysal/esda/blob/887058ffd36353744d5c7c52aed04222456f6380/pyproject.toml#L31).

Will update [dependency versions](https://github.com/pysal/esda/blob/887058ffd36353744d5c7c52aed04222456f6380/pyproject.toml#L32-L40) in follow-up PR.